ERROR - 2023-05-24 03:13:43 --> Error connecting to the database: mysqli_sql_exception: No connection could be made because the target machine actively refused it in C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\MySQLi\Connection.php:174
Stack trace:
#0 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\MySQLi\Connection.php(174): mysqli->real_connect('localhost', 'root', '', 'bookstore', 3306, '', 0)
#1 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseConnection.php(391): CodeIgniter\Database\MySQLi\Connection->connect(false)
#2 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseConnection.php(573): CodeIgniter\Database\BaseConnection->initialize()
#3 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseBuilder.php(1616): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#4 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Model.php(268): CodeIgniter\Database\BaseBuilder->get()
#5 C:\xampp\htdocs\ci4-uts\uts_ci4\system\BaseModel.php(656): CodeIgniter\Model->doFirst()
#6 C:\xampp\htdocs\ci4-uts\uts_ci4\app\Controllers\Auth.php(76): CodeIgniter\BaseModel->first()
#7 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(934): App\Controllers\Auth->login()
#8 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#9 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#10 C:\xampp\htdocs\ci4-uts\uts_ci4\public\index.php(67): CodeIgniter\CodeIgniter->run()
#11 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Commands\Server\rewrite.php(47): require_once('C:\\xampp\\htdocs...')
#12 {main}

Next CodeIgniter\Database\Exceptions\DatabaseException: No connection could be made because the target machine actively refused it in C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\MySQLi\Connection.php:218
Stack trace:
#0 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseConnection.php(391): CodeIgniter\Database\MySQLi\Connection->connect(false)
#1 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseConnection.php(573): CodeIgniter\Database\BaseConnection->initialize()
#2 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseBuilder.php(1616): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#3 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Model.php(268): CodeIgniter\Database\BaseBuilder->get()
#4 C:\xampp\htdocs\ci4-uts\uts_ci4\system\BaseModel.php(656): CodeIgniter\Model->doFirst()
#5 C:\xampp\htdocs\ci4-uts\uts_ci4\app\Controllers\Auth.php(76): CodeIgniter\BaseModel->first()
#6 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(934): App\Controllers\Auth->login()
#7 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#8 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#9 C:\xampp\htdocs\ci4-uts\uts_ci4\public\index.php(67): CodeIgniter\CodeIgniter->run()
#10 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Commands\Server\rewrite.php(47): require_once('C:\\xampp\\htdocs...')
#11 {main}
CRITICAL - 2023-05-24 03:13:43 --> Unable to connect to the database.
Main connection [MySQLi]: No connection could be made because the target machine actively refused it
in SYSTEMPATH\Database\BaseConnection.php on line 427.
 1 SYSTEMPATH\Database\BaseConnection.php(573): CodeIgniter\Database\BaseConnection->initialize()
 2 SYSTEMPATH\Database\BaseBuilder.php(1616): CodeIgniter\Database\BaseConnection->query('SELECT *
FROM `users`
WHERE `username` = :username:
 LIMIT 1', [...], false)
 3 SYSTEMPATH\Model.php(268): CodeIgniter\Database\BaseBuilder->get()
 4 SYSTEMPATH\BaseModel.php(656): CodeIgniter\Model->doFirst()
 5 APPPATH\Controllers\Auth.php(76): CodeIgniter\BaseModel->first()
 6 SYSTEMPATH\CodeIgniter.php(934): App\Controllers\Auth->login()
 7 SYSTEMPATH\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 8 SYSTEMPATH\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('FCPATH\\index.php')
ERROR - 2023-05-24 11:58:30 --> Error connecting to the database: mysqli_sql_exception: No connection could be made because the target machine actively refused it in C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\MySQLi\Connection.php:174
Stack trace:
#0 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\MySQLi\Connection.php(174): mysqli->real_connect('localhost', 'root', '', 'bookstore', 3306, '', 0)
#1 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseConnection.php(391): CodeIgniter\Database\MySQLi\Connection->connect(false)
#2 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseConnection.php(573): CodeIgniter\Database\BaseConnection->initialize()
#3 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseBuilder.php(1616): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#4 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Model.php(268): CodeIgniter\Database\BaseBuilder->get()
#5 C:\xampp\htdocs\ci4-uts\uts_ci4\system\BaseModel.php(656): CodeIgniter\Model->doFirst()
#6 C:\xampp\htdocs\ci4-uts\uts_ci4\app\Controllers\Auth.php(76): CodeIgniter\BaseModel->first()
#7 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(934): App\Controllers\Auth->login()
#8 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#9 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#10 C:\xampp\htdocs\ci4-uts\uts_ci4\public\index.php(67): CodeIgniter\CodeIgniter->run()
#11 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Commands\Server\rewrite.php(47): require_once('C:\\xampp\\htdocs...')
#12 {main}

Next CodeIgniter\Database\Exceptions\DatabaseException: No connection could be made because the target machine actively refused it in C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\MySQLi\Connection.php:218
Stack trace:
#0 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseConnection.php(391): CodeIgniter\Database\MySQLi\Connection->connect(false)
#1 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseConnection.php(573): CodeIgniter\Database\BaseConnection->initialize()
#2 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Database\BaseBuilder.php(1616): CodeIgniter\Database\BaseConnection->query('SELECT *\nFROM `...', Array, false)
#3 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Model.php(268): CodeIgniter\Database\BaseBuilder->get()
#4 C:\xampp\htdocs\ci4-uts\uts_ci4\system\BaseModel.php(656): CodeIgniter\Model->doFirst()
#5 C:\xampp\htdocs\ci4-uts\uts_ci4\app\Controllers\Auth.php(76): CodeIgniter\BaseModel->first()
#6 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(934): App\Controllers\Auth->login()
#7 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
#8 C:\xampp\htdocs\ci4-uts\uts_ci4\system\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(NULL, Object(Config\Cache), false)
#9 C:\xampp\htdocs\ci4-uts\uts_ci4\public\index.php(67): CodeIgniter\CodeIgniter->run()
#10 C:\xampp\htdocs\ci4-uts\uts_ci4\system\Commands\Server\rewrite.php(47): require_once('C:\\xampp\\htdocs...')
#11 {main}
CRITICAL - 2023-05-24 11:58:30 --> Unable to connect to the database.
Main connection [MySQLi]: No connection could be made because the target machine actively refused it
in SYSTEMPATH\Database\BaseConnection.php on line 427.
 1 SYSTEMPATH\Database\BaseConnection.php(573): CodeIgniter\Database\BaseConnection->initialize()
 2 SYSTEMPATH\Database\BaseBuilder.php(1616): CodeIgniter\Database\BaseConnection->query('SELECT *
FROM `users`
WHERE `username` = :username:
 LIMIT 1', [...], false)
 3 SYSTEMPATH\Model.php(268): CodeIgniter\Database\BaseBuilder->get()
 4 SYSTEMPATH\BaseModel.php(656): CodeIgniter\Model->doFirst()
 5 APPPATH\Controllers\Auth.php(76): CodeIgniter\BaseModel->first()
 6 SYSTEMPATH\CodeIgniter.php(934): App\Controllers\Auth->login()
 7 SYSTEMPATH\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Auth))
 8 SYSTEMPATH\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('FCPATH\\index.php')
CRITICAL - 2023-05-24 12:29:08 --> Attempt to read property "firstName" on array
in APPPATH\Views\templates\header.php on line 43.
 1 APPPATH\Views\templates\header.php(43): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Attempt to read property "firstName" on array', 'APPPATH\\Views\\templates\\header.php', 43)
 2 SYSTEMPATH\View\View.php(213): include('APPPATH\\Views\\templates\\header.php')
 3 SYSTEMPATH\View\View.php(216): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 4 SYSTEMPATH\Common.php(1184): CodeIgniter\View\View->render('templates/header', [], true)
 5 APPPATH\Controllers\Main.php(93): view('templates/header', [...])
 6 SYSTEMPATH\CodeIgniter.php(934): App\Controllers\Main->employee()
 7 SYSTEMPATH\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Main))
 8 SYSTEMPATH\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('FCPATH\\index.php')
CRITICAL - 2023-05-24 12:32:19 --> Undefined variable $list
in APPPATH\Views\templates\header.php on line 43.
 1 APPPATH\Views\templates\header.php(43): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Undefined variable $list', 'APPPATH\\Views\\templates\\header.php', 43)
 2 SYSTEMPATH\View\View.php(213): include('APPPATH\\Views\\templates\\header.php')
 3 SYSTEMPATH\View\View.php(216): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 4 SYSTEMPATH\Common.php(1184): CodeIgniter\View\View->render('templates/header', [], true)
 5 APPPATH\Controllers\Pos.php(29): view('templates/header', [...])
 6 SYSTEMPATH\CodeIgniter.php(934): App\Controllers\Pos->cashier()
 7 SYSTEMPATH\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Pos))
 8 SYSTEMPATH\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('FCPATH\\index.php')
CRITICAL - 2023-05-24 12:34:59 --> Undefined variable $ses_data
in APPPATH\Views\templates\header.php on line 43.
 1 APPPATH\Views\templates\header.php(43): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Undefined variable $ses_data', 'APPPATH\\Views\\templates\\header.php', 43)
 2 SYSTEMPATH\View\View.php(213): include('APPPATH\\Views\\templates\\header.php')
 3 SYSTEMPATH\View\View.php(216): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 4 SYSTEMPATH\Common.php(1184): CodeIgniter\View\View->render('templates/header', [], true)
 5 APPPATH\Controllers\Main.php(93): view('templates/header', [...])
 6 SYSTEMPATH\CodeIgniter.php(934): App\Controllers\Main->employee()
 7 SYSTEMPATH\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Main))
 8 SYSTEMPATH\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('FCPATH\\index.php')
CRITICAL - 2023-05-24 12:35:15 --> Cannot use object of type CodeIgniter\Session\Session as array
in APPPATH\Views\templates\header.php on line 43.
 1 SYSTEMPATH\View\View.php(213): include()
 2 SYSTEMPATH\View\View.php(216): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 3 SYSTEMPATH\Common.php(1184): CodeIgniter\View\View->render('templates/header', [], true)
 4 APPPATH\Controllers\Main.php(93): view('templates/header', [...])
 5 SYSTEMPATH\CodeIgniter.php(934): App\Controllers\Main->employee()
 6 SYSTEMPATH\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Main))
 7 SYSTEMPATH\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 8 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
 9 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('FCPATH\\index.php')
CRITICAL - 2023-05-24 12:36:07 --> Undefined variable $ses_data
in APPPATH\Views\templates\header.php on line 43.
 1 APPPATH\Views\templates\header.php(43): CodeIgniter\Debug\Exceptions->errorHandler(2, 'Undefined variable $ses_data', 'APPPATH\\Views\\templates\\header.php', 43)
 2 SYSTEMPATH\View\View.php(213): include('APPPATH\\Views\\templates\\header.php')
 3 SYSTEMPATH\View\View.php(216): CodeIgniter\View\View->CodeIgniter\View\{closure}()
 4 SYSTEMPATH\Common.php(1184): CodeIgniter\View\View->render('templates/header', [], true)
 5 APPPATH\Controllers\Main.php(93): view('templates/header', [...])
 6 SYSTEMPATH\CodeIgniter.php(934): App\Controllers\Main->employee()
 7 SYSTEMPATH\CodeIgniter.php(499): CodeIgniter\CodeIgniter->runController(Object(App\Controllers\Main))
 8 SYSTEMPATH\CodeIgniter.php(368): CodeIgniter\CodeIgniter->handleRequest(null, Object(Config\Cache), false)
 9 FCPATH\index.php(67): CodeIgniter\CodeIgniter->run()
10 SYSTEMPATH\Commands\Server\rewrite.php(47): require_once('FCPATH\\index.php')
